proto/src/determined/resourcepool/v1/resourcepool.proto

```diff
@@ -0,0 +1,182 @@
+  string image_id=3;
+  // Key for tagging the Determined agent instances
+  string tag_key=4;
+  // Value for tagging the Determined agent instances
+  string tag_value=5;
+  // Name to set for the Determined agent instances
+  string instance_name=6;
+  // The name of the SSH key registered with AWS for SSH key access to the agent instances
+  string ssh_key_name=7;
+  // Whether to use public IP addresses for the Determined agent
+  bool public_ip=8;
+  // The ID of the subnet to run the Determined agents in
+  string subnet_id=9;
+  // The ID of the security group to run the Determined agents as
+  string security_group_id=10;
+  // The Amazon Resource Name (ARN) of the IAM instance profile to attach to the agent instances.
+  string iam_instance_profile_arn=11;
+  // AWS instance type to use for dynamic agents
+  string instance_type=12;
+  // The log group
+  string log_group=13;
+  // The log stream
+  string log_stream=14;
+  // Whether to use spot instances
+  bool spot_enabled=15;
+  //  The maximum price per hour to pay for a spot instance
+  string spot_max_price=16;
+  // List of arbitrary user-defined tags that are added to the Determined agent instances
+  repeated determined.resourcepool.v1.AwsCustomTag custom_tags=17;
+}
+
+// GCP-specific details about the resource pool
+message ResourcePoolGcpDetail {
+  // The project ID of the GCP resources used by Determined
+  string project=1;
+  // The zone of the GCP resources used by Determined
+  string zone=2;
+  // Size of the root volume of the Determined agent in GB
+  int32 boot_disk_size=3;
+  // The boot disk source image of the Determined agent
+  string boot_disk_source_image=4;
+  // Key for labeling the Determined agent instances.
+  string label_key=5;
+  //  Value for labeling the Determined agent instances
+  string label_value=6;
+  //  Name prefix to set for the Determined agent instances
+  string name_prefix=7;
+  // Network resource for the Determined agent instances
+  string network=8;
+  // Subnetwork resource for the Determined agent instances
+  string subnetwork=9;
+  // Whether to use external IP addresses for the Determined agent instances
+  bool external_ip=10;
+  // The network tags to set firewalls for the Determined agent instances
+  repeated string network_tags=11;
+  //  Email of the service account for the Determined agent instances.
+  string service_account_email=12;
+  //  List of scopes authorized for the Determined agent instances
+  repeated string service_account_scopes=13;
+  // Type of machine for the Determined agents
+  string machine_type=14;
+  // Type of GPU for the Determined agents
+  string gpu_type=15;
+  // Number of GPUs for the Determined agents
+  int32 gpu_num=16;
+  // Whether to use preemptible instances
+  bool preemptible=17;
+  // The timeout period for tracking a GCP operation
+  string operation_timeout_period=18;
+}
+
+// Details related to the priority scheduler. This will only be present if the schedulerType=priority
```

Comment:
q: do we just have a detail struct for priority scheduler? why not for fair share and round robin?

Reply:
Yeah, only the priority scheduler can be configured.